middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -19,13 +21,23 @@ def https?
       # Start an instance of Middleman::Application
       # @return [void]
       def start(opts={})
+        # Otherwise testing becomes hard
+        $stdout.sync = true
+        $stderr.sync = true
+
         @options = opts
 
+        @server_information = ServerInformation.new(
+          bind_address: opts[:bind_address],
+          server_name: opts[:server_name],
+        )
+
         mount_instance(new_app)
 
         logger.debug %(== The Middleman is running in "#{environment}" environment)
-        logger.info "== The Middleman is standing watch at #{uri} (#{uri(public_ip)})"
-        logger.info "== Inspect your site configuration at #{uri + '__middleman'}"
```

Comment:
* There are listeners, that's where `netstat -nlplt` (gnu netstat) gives you listening services.
* Normal Urls to open in Browser
* Config urls to open in Browser